angular/projects/lib/src/lib/base-widget.ts

```diff
@@ -20,17 +20,25 @@ import { NgCompInputs, NgGridStackWidget } from './types';
    * REDEFINE to return an object representing the data needed to re-create yourself, other than `selector` already handled.
    * This should map directly to the @Input() fields of this objects on create, so a simple apply can be used on read
    */
-  public serialize(): NgCompInputs | undefined  { return; }
+  public serialize(): NgCompInputs | undefined { return; }
 
   /**
    * REDEFINE this if your widget needs to read from saved data and transform it to create itself - you do this for
    * things that are not mapped directly into @Input() fields for example.
    */
-  public deserialize(w: NgGridStackWidget)  {
-    // save full description for meta data
+  public deserialize(w: NgGridStackWidget) {
     this.widgetItem = w;
-    if (!w) return;
 
-    if (w.input)  Object.assign(this, w.input);
+    if (!w?.input) return;
+
+    for (const [key, value] of Object.entries(w.input)) {
+      const prop = (this as any)[key];
+
+      if (typeof prop === 'function' && 'set' in prop) {
+        (prop as WritableSignal<unknown>).set(value);

```

Comment:
could this will be done in older angular (say 14 we have today. I didn't check when signal came in...